pkg/virt-operator/kubevirt.go

```diff
@@ -659,17 +661,30 @@ func (c *KubeVirtController) execute(key string) error {
 	// set timestamps on conditions if they changed
 	operatorutil.SetConditionTimestamps(kv, kvCopy)
 
-	// If we detect a change on KubeVirt we update it
+	// If we detect a change on KubeVirt status we update it
 	if !reflect.DeepEqual(kv.Status, kvCopy.Status) {
-		if err := c.statusUpdater.UpdateStatus(kvCopy); err != nil {
-			logger.Reason(err).Errorf("Could not update the KubeVirt resource status.")
+		statusJson, err := json.Marshal(kvCopy.Status)
+		if err != nil {
+			return err
+		}
+		patch := fmt.Sprintf(`[{"op": "replace", "path": "/status", "value": %s}]`, string(statusJson))
+		err = c.statusUpdater.PatchStatus(kv, types.JSONPatchType, []byte(patch))
+		if err != nil {
+			logger.Reason(err).Errorf("Could not patch the KubeVirt resource status.")
 			return err
 		}
 	}
 
+	// If we detect a change on KubeVirt finalizers we update them
 	if !reflect.DeepEqual(kv.Finalizers, kvCopy.Finalizers) {
-		if _, err := c.clientset.KubeVirt(kvCopy.ObjectMeta.Namespace).Update(kvCopy); err != nil {
-			logger.Reason(err).Errorf("Could not update the KubeVirt resource.")
+		finalizersJson, err := json.Marshal(kvCopy.Finalizers)
+		if err != nil {
+			return err
+		}
+		patch := fmt.Sprintf(`[{"op": "replace", "path": "/metadata/finalizers", "value": %s}]`, string(finalizersJson))
```

Comment:
Here patch makes a lot of sense :+1: . We just want to ensure that a certion field is set and don't care about conflicts.